zepid/calc/utils.py

```diff
@@ -1,42 +1,89 @@
 import warnings
-import math
+from typing import NamedTuple
+
+import numpy as np
 from scipy.stats import norm
 
 
-def risk_ci(events, total, alpha=0.05, confint='wald'):
-    """Calculate two-sided (1-alpha)% Confidence interval of Risk. Note
-    relies on the Central Limit Theorem, so there must be at least 5 events
-    and 5 nonevents
+class Results(NamedTuple):
+    """
+    A simple wrapper class for the results of marginal estimates
+    """
+    point_estimate: float
+    lower_bound: float
+    upper_bound: float
+    standard_error: float
+    alpha: float
+    name: str
 
-    Returns (risk, lower CL, upper CL, SE)
 
-    events:
-        -Number of events/outcomes that occurred
-    total:
-        -total number of subjects that could have experienced the event
-    alpha:
-        -Alpha level. Default is 0.05
-    confint:
-        -Type of confidence interval to generate. Current options are
-            wald
-            hypergeometric
+def normal_ppf(z):
```

Comment:
oh yea abstracted this

Reply:
I like this abstraction. Especially the checks. Might have to consider a similar approach in other branches (at least for ``norm.ppf``